pkgs/development/compilers/idris2/default.nix

```diff
@@ -1,96 +1,83 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, makeWrapper
-, clang
+{ stdenv
+, lib
 , chez
+, chez-racket
+, clang
 , gmp
+, fetchFromGitHub
+, makeWrapper
+, gambit
+, nodejs
 , zsh
 , callPackage
 }:
 
-# NOTICE: An `idris2WithPackages` is available at: https://github.com/claymager/idris2-pkgs
-
-# Uses scheme to bootstrap the build of idris2
-stdenv.mkDerivation rec {
-  pname = "idris2";
-  version = "0.5.1";
+let
+  idris2-version = "0.5.1";  # When updating this, remove postInstall override below
 
-  src = fetchFromGitHub {
+  idris2-src = fetchFromGitHub {
     owner = "idris-lang";
     repo = "Idris2";
-    rev = "v${version}";
+    rev = "v${idris2-version}";
     sha256 = "sha256-6CTn8o5geWSesXO7vTrrV/2EOQ3f+nPQ2M5cem13ZSY=";
   };
 
-  # We do not add any propagatedNativeBuildInputs because we do not want the
-  # executables idris2 produces to depend on the nix-store. As such, it is left
-  # to the user to guarantee chez (or any other codgen dependency) is available
-  # in the path during compilation of programs with idris2.
-  strictDeps = true;
-  nativeBuildInputs = [ makeWrapper clang chez ]
-    ++ lib.optional stdenv.isDarwin [ zsh ];
-  buildInputs = [ gmp ];
-
-  prePatch = ''
-    patchShebangs --build tests
-  '';
-
-  makeFlags = [ "PREFIX=$(out)" ]
-    ++ lib.optional stdenv.isDarwin "OS=";
-
-  # The name of the main executable of pkgs.chez is `scheme`
-  buildFlags = [ "bootstrap" "SCHEME=scheme" ];
-
-  checkTarget = "test";
-
-  # TODO: Move this into its own derivation, such that this can be changed
-  #       without having to recompile idris2 every time.
-  postInstall = let
-    includedLibs = [ "base" "contrib" "network" "prelude" ];
-    name = "${pname}-${version}";
-    packagePaths =
-      builtins.map (l: "$out/${name}/${l}-${version}") includedLibs;
-    additionalIdris2Paths = builtins.concatStringsSep ":" packagePaths;
-  in ''
-    # Remove existing idris2 wrapper that sets incorrect LD_LIBRARY_PATH
-    rm $out/bin/idris2
-    # Move actual idris2 binary
-    mv $out/bin/idris2_app/idris2.so $out/bin/idris2
-
-    # After moving the binary, there is nothing left in idris2_app that isn't
-    # either contained in lib/ or is useless to us.
-    rm $out/bin/idris2_app/*
-    rmdir $out/bin/idris2_app
+  idris2-derivation = callPackage "${idris2-src}/nix/package.nix" {
```

Comment:
We cannot do this in nixpkgs as we don't allow import-from-derivation for performance reason.

Reply:
Perfect, that makes sense. Thanks for taking a look and explaining the reasoning.

Reply:
basically everyone that would want to evaluate idris2 i.e. when evaluating all packages locally with nixpkgs-review, would also need to download the source code of idris2, which is not desirable (if we do this with other packages as well).